### ISSUE TITLE
Use a forward proxy to mock apis inside containers created by `act`

### ISSUE THREAD

**Issue by shubhbapna** [kiegroup/mock-github#47]
## Idea
When launching a container, it is possible to specify a forward proxy by setting env variables - `HTTP_PROXY, http_proxy, HTTPS_PROXY, https_proxy` [1].

Moreover, `act` allows us to pass env variables [2], you we can easily add a forward proxy to any container created by `act`. 

I did some quick checking by running a super quick TCP server using netcat and passing the required env variables to `act` when running a sample workflow. Any http/https calls from the workflow were in fact sent to the tcp server. So I think this method has potential.

The idea is to pass what apis you want to mock directly to `act`. We will run a forward proxy server before running `act` and set the env vars to point to that forward proxy. The forward proxy on receiving request will determine whether the request has to be mocked or not. If it is mocked it will send back the user response otherwise it will forward the call to the actual host.
```typescript
const moctokit = new Moctokit();
const mockapi = new Mockapi(schema);
const act = new Act();
act.runEvent("push", {
      mockApi: [
             moctokit.rest.repos.get().setResponse({status: 200, data: {msg: "hello"}),
             mockapi.mock.customApi.root.search().setRespone({status: 200, data: {msg: "world"})
      ]
})
```
## Impact
- This will deprecate `ActionCompiler` since the APIs are going to be directly mocked in the container. We won't have to inject mocks into the custom actions.
- Users can in fact run the workflows as is even when they don't have access to the code that the action is going to run i.e. they are not able to inject mocks into the action. They can simply mock whatever api they want inside the container

[1] https://docs.docker.com/network/proxy/#use-environment-variables
[2] https://github.com/nektos/act#flags

**Comment by Ginxo** [comment]
this makes a lot of sense and will allow to really mock whatever the http request. We should proceed this way. Congrats @shubhbapna and many thanks!

**Comment by shubhbapna** [comment]
HTTP works very nicely but HTTPS is being problematic. For any HTTPS connection originating from within the container, the container first sends a CONNECT request to the proxy. Now usually the job of the proxy is to setup a connection to the final destination and then pipe it to the connection it received from the container. However, once that is done the proxy is not able to see the actual request. I am not seeing any work around without having to implement some sort of a MITM

**Comment by shubhbapna** [comment]
Resolved in kiegroup/act-js#2
